core/src/main/java/com/linecorp/armeria/server/AnnotatedValueResolver.java

```diff
@@ -683,10 +766,16 @@ private Builder(AnnotatedElement annotatedElement, Type type) {
         }
 
         /**
-         * Sets the annotation which is one of {@link Param}, {@link Header} or {@link RequestObject}.
+         * Sets the annotation type which is one of the following.
+         * <ul>
+         *     <li>{@link Param}</li>
+         *     <li>{@link Header}</li>
+         *     <li>{@link RequestBean}</li>
+         *     <li>{@link RequestConverter}</li>
+         * </ul>
          */
-        private Builder annotation(Annotation annotation) {
-            this.annotation = annotation;
+        private Builder annotationType(Class<? extends Annotation> annotationType) {
+            this.annotationType = annotationType;
```

Comment:
I'm curious if it's a good idea to add an assertion here to make sure `annotationType` is one of the four types.